integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -0,0 +1,1361 @@
+      });
+
+      it("throws an error if the Realm does not have a sync config", function (this: RealmContext) {
+        const realm = new Realm({ schema: [FlexiblePersonSchema] });
+        expect(() => realm.subscriptions).to.throw(
+          "`subscriptions` can only be accessed if flexible sync is enabled, but sync is currently disabled for your app. Add a flexible sync config when opening the Realm, for example: { sync: { user, flexible: true } }",
+        );
+      });
+
+      it("throws an error if the Realm has a partition based sync config", function (this: RealmContext) {
+        const realm = new Realm({ schema: [FlexiblePersonSchema], sync: { user: this.user, partitionValue: "test" } });
+        expect(() => realm.subscriptions).to.throw(
+          "`subscriptions` can only be accessed if flexible sync is enabled, but partition based sync is currently enabled for your Realm. Modify your sync config to remove any `partitionValue` and enable flexible sync, for example: { sync: { user, flexible: true } }",
+        );
+      });
+    });
+
+    describe("Subscription class", function () {
+      it("has an id", function (this: RealmContext) {
+        const sub = addSubscriptionForPerson(this.realm).sub;
+        expect(sub.id).to.be.instanceOf(BSON.ObjectId);
+      });
+
+      it("has a createdAt date", function (this: RealmContext) {
+        const sub = addSubscriptionForPerson(this.realm).sub;
+        expect(sub.createdAt).to.be.instanceOf(Date);
+      });
+
+      it("has an updatedAt date", function (this: RealmContext) {
+        const sub = addSubscriptionForPerson(this.realm).sub;
+        expect(sub.updatedAt).to.be.instanceOf(Date);
+      });
+
+      it("has a default name", function (this: RealmContext) {
+        const sub = addSubscriptionForPerson(this.realm).sub;
+        expect(sub.name).to.equal(null);
+      });
+
+      it("has a specified name", function (this: RealmContext) {
+        const subWithName = addSubscriptionForPerson(this.realm, { name: "test" }).sub;
+        expect(subWithName.name).to.equal("test");
+      });
+
+      it("has an objectType", function (this: RealmContext) {
+        const sub = addSubscriptionForPerson(this.realm).sub;
+        expect(sub.objectType).to.equal(FlexiblePersonSchema.name);
+      });
+
+      it("has a default queryString", function (this: RealmContext) {
+        const sub = addSubscriptionForPerson(this.realm).sub;
+        expect(sub.queryString).to.equal("TRUEPREDICATE");
+      });
+
+      it("has a specified queryString", function (this: RealmContext) {
+        const subWithFilter = addSubscription(
+          this.realm,
+          this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10"),
+        ).sub;
+        expect(subWithFilter.queryString).to.equal("age > 10");
+      });
+
+      it("contains interpolated arguments in the queryString", function (this: RealmContext) {
```

Comment:
You get something like `friends == O45` , where `On` is incrementing with each test run - so I guess is a representation of the internal primary key or something? Probably not a great user facing thing but I'm not sure what the correct behaviour would be otherwise. Again, will raise for discussion/